drivers/regulator/regulator_axp2101.c

```diff
@@ -0,0 +1,463 @@
+	.vsel_mask = 0x1FU,
+	.vsel_bitpos = 0U,
+	.max_ua = 300000U,
+	.ranges = bldo1_ranges,
+	.num_ranges = ARRAY_SIZE(bldo1_ranges),
+};
+
+static const struct linear_range bldo2_ranges[] = {
+	LINEAR_RANGE_INIT(500000U, 100000U, 0x00U, 0x1EU),
+};
+
+__maybe_unused static const struct regulator_axp2101_desc bldo2_desc = {
+	.enable_reg = AXP2101_REG_LDOGRP1_CONTROL,
+	.enable_mask = 0x20U,
+	.enable_val = 0x20U,
+	.vsel_reg = AXP2101_REG_BLDO2_VOLTAGE,
+	.vsel_mask = 0x1FU,
+	.vsel_bitpos = 0U,
+	.max_ua = 300000U,
+	.ranges = bldo2_ranges,
+	.num_ranges = ARRAY_SIZE(bldo2_ranges),
+};
+
+static const struct linear_range cpusldo_ranges[] = {
+	LINEAR_RANGE_INIT(500000U, 50000U, 0x00U, 0x13U),
+};
+
+__maybe_unused static const struct regulator_axp2101_desc cpusldo_desc = {
+	.enable_reg = AXP2101_REG_LDOGRP1_CONTROL,
+	.enable_mask = 0x40U,
+	.enable_val = 0x40U,
+	.vsel_reg = AXP2101_REG_CPUSLDO_VOLTAGE,
+	.vsel_mask = 0x1FU,
+	.vsel_bitpos = 0U,
+	.max_ua = 30000U,
+	.ranges = cpusldo_ranges,
+	.num_ranges = ARRAY_SIZE(cpusldo_ranges),
+};
+
+static const struct linear_range dldo1_ranges[] = {
+	LINEAR_RANGE_INIT(500000U, 100000U, 0x00U, 0x1CU),
+};
+
+__maybe_unused static const struct regulator_axp2101_desc dldo1_desc = {
+	.enable_reg = AXP2101_REG_LDOGRP1_CONTROL,
+	.enable_mask = 0x80U,
+	.enable_val = 0x80U,
+	.vsel_reg = AXP2101_REG_DLDO1_VOLTAGE,
+	.vsel_mask = 0x1FU,
+	.vsel_bitpos = 0U,
+	.max_ua = 300000U,
+	.ranges = dldo1_ranges,
+	.num_ranges = ARRAY_SIZE(dldo1_ranges),
+};
+
+static const struct linear_range dldo2_ranges[] = {
+	LINEAR_RANGE_INIT(500000U, 100000U, 0x00U, 0x13U),
+};
+
+__maybe_unused static const struct regulator_axp2101_desc dldo2_desc = {
+	.enable_reg = AXP2101_REG_LDOGRP2_CONTROL,
+	.enable_mask = 0x01U,
+	.enable_val = 0x01U,
+	.vsel_reg = AXP2101_REG_DLDO2_VOLTAGE,
+	.vsel_mask = 0x1FU,
+	.vsel_bitpos = 0U,
+	.max_ua = 300000U,
+	.ranges = dldo2_ranges,
+	.num_ranges = ARRAY_SIZE(dldo2_ranges),
+};
+
+static int axp2101_enable(const struct device *dev)
+{
+	const struct regulator_axp2101_config *config = dev->config;
+	int ret;
+
+	LOG_INST_DBG(config->log, "Enabling regulator");
+	LOG_INST_DBG(config->log, "[0x%02x]=0x%02x mask=0x%02x", config->desc->enable_reg,
+		     config->desc->enable_val, config->desc->enable_mask);
+
+	ret = i2c_reg_update_byte_dt(&config->i2c, config->desc->enable_reg,
+				     config->desc->enable_mask, config->desc->enable_val);
+
+	if (ret != 0) {
+		LOG_INST_ERR(config->log, "Failed to enable regulator");
+	}
+
+	return ret;
+}
+
+static int axp2101_disable(const struct device *dev)
+{
+	const struct regulator_axp2101_config *config = dev->config;
+	int ret;
+
+	LOG_INST_DBG(config->log, "Disabling regulator");
+	LOG_INST_DBG(config->log, "[0x%02x]=0 mask=0x%x", config->desc->enable_reg,
+		     config->desc->enable_mask);
+
+	ret = i2c_reg_update_byte_dt(&config->i2c, config->desc->enable_reg,
+				     config->desc->enable_mask, 0u);
+	if (ret != 0) {
```

Comment:
```suggestion
	if (ret < 0) {
```

others as well

Reply:
Hi @gmarull 
thanks, I changed the if statements